pina/problem/abstract_problem.py

```diff
@@ -41,26 +38,33 @@ def __init__(self):
                     self.domains[cond_name] = cond.domain
                     cond.domain = cond_name
 
+        self._collect_data = {}
+
     @property
-    def batching_dimension(self):
+    def collected_data(self):
         """
-        Get batching dimension.
+        Return the collected data from the problem's conditions.
 
-        :return: The batching dimension.
-        :rtype: int
+        :return: The collected data.
+        :rtype: dict
         """
-        return self._batching_dimension
+        if not self._collect_data:
+            raise RuntimeError(
+                "You have to call collect_data() before accessing the data."
+            )
+        return self._collect_data
 
-    @batching_dimension.setter
-    def batching_dimension(self, value):
+    @collected_data.setter
```

Comment:
Do you use it? I would avoid to have a setter, otherwise the user can change it easily.